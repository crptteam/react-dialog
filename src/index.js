import Dialog from './components/Dialog';

export { Dialog };

export default Dialog;