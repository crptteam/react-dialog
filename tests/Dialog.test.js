import React from "react";
import { Dialog } from "../src/";
import { mount } from "enzyme";

describe("Dialog", () => {
  it("Should renders without problems", () => {
    const wrapper = mount(
      <Dialog
        isOpen={true}
        title={'Изменение настроек'}
        content={<div>Вы изменили настройки учетной записи.<br/>
          Вы уверены, что хотите не сохранять их?</div>}
        buttons={[
          {
            children: 'Отменить изменения',
            onClick: e => console.log('Отменить изменения')
          },
          {
            children: 'Сохранить изменения',
            onClick: e => console.log('Сохранить изменения')
          }
        ]}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
