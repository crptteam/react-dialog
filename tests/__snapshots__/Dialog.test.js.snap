// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Should renders without problems 1`] = `
<WithTheme(Dialog)
  buttons={
    Array [
      Object {
        "children": "Отменить изменения",
        "onClick": [Function],
      },
      Object {
        "children": "Сохранить изменения",
        "onClick": [Function],
      },
    ]
  }
  content={
    <div>
      Вы изменили настройки учетной записи.
      <br />
      Вы уверены, что хотите не сохранять их?
    </div>
  }
  isOpen={true}
  title="Изменение настроек"
>
  <Dialog
    buttons={
      Array [
        Object {
          "children": "Отменить изменения",
          "onClick": [Function],
        },
        Object {
          "children": "Сохранить изменения",
          "onClick": [Function],
        },
      ]
    }
    content={
      <div>
        Вы изменили настройки учетной записи.
        <br />
        Вы уверены, что хотите не сохранять их?
      </div>
    }
    isOpen={true}
    theme={
      Object {
        "Tooltip": Object {},
      }
    }
    title="Изменение настроек"
  >
    <Dialog__Overlay
      show={true}
    >
      <div
        className="Dialog__Overlay-hZaxQJ btOXTY"
      >
        <Dialog__Wrap>
          <div
            className="Dialog__Wrap-hZvJaQ xGLtP"
          >
            <Dialog__Header>
              <div
                className="Dialog__Header-flZyoF iOTdGp"
              >
                Изменение настроек
              </div>
            </Dialog__Header>
            <Dialog__Content>
              <div
                className="Dialog__Content-dxmOQa VTJeA"
              >
                <div>
                  Вы изменили настройки учетной записи.
                  <br />
                  Вы уверены, что хотите не сохранять их?
                </div>
              </div>
            </Dialog__Content>
            <Dialog__ButtonsWrap>
              <div
                className="Dialog__ButtonsWrap-crANOr bujAJe"
              >
                <Dialog__Button
                  key="0"
                  onClick={[Function]}
                >
                  <div
                    className="Dialog__Button-liAEUe kKICwr"
                    onClick={[Function]}
                  >
                    Отменить изменения
                  </div>
                </Dialog__Button>
                <Dialog__Button
                  key="1"
                  onClick={[Function]}
                >
                  <div
                    className="Dialog__Button-liAEUe kKICwr"
                    onClick={[Function]}
                  >
                    Сохранить изменения
                  </div>
                </Dialog__Button>
              </div>
            </Dialog__ButtonsWrap>
          </div>
        </Dialog__Wrap>
      </div>
    </Dialog__Overlay>
  </Dialog>
</WithTheme(Dialog)>
`;
